### ISSUE TITLE
Bug: Wrong commit count

### ISSUE THREAD

**Issue by anshumanv** [anuraghazra/github-readme-stats#14]
Hey 👋 

Cool project, is there a limitation to the number of commits/PRs it can fetch?

They seem to be off by quite a lot for instance - 

![image](https://user-images.githubusercontent.com/21009455/87129089-6acae000-c2ae-11ea-8edc-b46abce403a2.png)

In normal GitHub search - 
![image](https://user-images.githubusercontent.com/21009455/87129136-7d451980-c2ae-11ea-8033-2674ad204b62.png)

**Comment by anuraghazra** [comment]
Hmm it should be fine.. it fetches the data from Github GraphQL api so it can't be invalid. although it is possible that you contributed to github organizations and i think those are not counted. #1 

![Stat](https://github-readme-stats.vercel.app/api?username=anshumanv)

**Comment by anshumanv** [comment]
Ah okay, very strange because PRs count is close to correct but commit count is still far off even if you exclude org commits.

**Comment by anuraghazra** [comment]
@anshumanv i think i got it. the issue is that github only returns last years contribution commits 

```gql

{
  user(login: "anshumanv") {
   contributionsCollection(from: "2016-08-13T14:31:00Z", to:"2020-07-10T08:35:06.094Z") {
      totalCommitContributions
    }
  }
}

```


I can set from and to variables in the query but github does not allow more than one year's contributions to be shown at once

I have to find a way to fix this. do you have any ideas on this?

**Comment by anuraghazra** [comment]
@anshumanv what i did is that looped through all the years and fetched the commits individually and seems kinda close. 

![github-readme-issue](https://user-images.githubusercontent.com/35374649/87137317-51c82c00-c2ba-11ea-926d-29daf764c8dc.PNG)

**Comment by anshumanv** [comment]
Yes, this is the basic brute force solution but I think you'll run into API rate limit after a while if you do this for every user.

**Comment by anshumanv** [comment]
Maybe you can give an option for users to use their own tokens?

**Comment by anuraghazra** [comment]
No i don't think i can hit the rate limit because it's already authenticated by the server... i auth token in env.

**Comment by anuraghazra** [comment]
Also i don't think i can give users an option to add their own tokens because that would be a huge security vulnrability if someone did like this `/api?username=abc&token=SECRET_TOKEN` because the url would be public

**Comment by anshumanv** [comment]
> No i don't think i can hit the rate limit because it's already authenticated by the server... i set auth token in env.

Yep even that doesn't give you infinite requests 😄 , IIRC you get 5k req per hour which can work for now till this gets really popular.

**Comment by anshumanv** [comment]
> Also i don't think i can give users an option to add their own tokens because that would be a huge security vulnrability if someone did like this /api?username=abc&token=SECRET_TOKEN because the url would be public

Yes, only option is for user to host their own app but that seems like an overkill for now

**Comment by anshumanv** [comment]
And anyways tokens in request params will be deprecated in near future. You have to send it in body.

**Comment by anuraghazra** [comment]
> > No i don't think i can hit the rate limit because it's already authenticated by the server... i set auth token in env.
> 
> Yep even that doesn't give you infinite requests  , IIRC you get 5k req per hour which can work for now till this gets really popular.

Just hope not everyone would have commits from 1980's like yours :laughing: your account will hit the api 42 times :laughing:

**Comment by anshumanv** [comment]
> Just hope not everyone would have commits from 1980's like yours 😆 your account will hit the api 42 times 😆

Lol, that was when in my 1st year I committed using a messed up system clock, forgot to revert it 😥 

Maybe I should take care of it real quick 😛

**Comment by 1UC1F3R616** [comment]
@anshumanv It's showing yearly commit for me I see not total from when I started
Is this a feture for future pr or I using some wrong parameter?

**Comment by anuraghazra** [comment]
@1UC1F3R616 it is currently shows 1years total commit, its a limitation of github api. checkout #15 for more info

**Comment by omidnikrah** [comment]
I think, at the moment it's better to change "Total Commits:" to "Total Commits (last year):".

**Comment by vasugamdha** [comment]
I don't know much, but is it possible that before people use it, they have to fork it, modify the auth token and then request? So like, in future you don't have to face rate limit of 5k!? I might even be wrong, conceptually.

**Comment by anuraghazra** [comment]
Hey @vasugamdha Let's discuss about rate limit on another issue maybe? you can open an issue to discuss it or hop into my discord :) https://discord.com/invite/VK4k3Br

**Comment by anuraghazra** [comment]
Fixed in #211
